tapdb/sqlc/queries/addrs.sql

```diff
@@ -1,9 +1,13 @@
--- name: InsertAddr :one
+-- name: UpsertAddr :one
 INSERT INTO addrs (
     version, asset_version, genesis_asset_id, group_key, script_key_id,
     taproot_key_id, tapscript_sibling, taproot_output_key, amount, asset_type,
     creation_time, proof_courier_addr
-) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12) RETURNING id;
+) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12)
+ON CONFLICT(taproot_output_key) DO UPDATE SET
+    -- no-op update to trigger RETURNING clause
+    taproot_output_key = excluded.taproot_output_key
+RETURNING id;
```

Comment:
If there's a conflict on the taproot output key, I think the query should update the proof courier address, since that's not part of the taproot output key.

Otherwise, I think this should be compatible with how `tapgarden/custodian` currently operates.

Reply:
Had a 1:1 with Oli. My idea to allow updating the proof courier address here isn't sound. If a user has already shared their Tap address, we shouldn’t allow them to modify it in the database afterward. They will look to the wrong courier for the proof.